From 5959bf5453ca80f8dad566c769bcb75e800000f9 Mon Sep 17 00:00:00 2001
From: Suryanarayana Raju Sangani <ssangani@xilinx.com>
Date: Wed, 5 Apr 2023 21:51:56 +0530
Subject: [PATCH] net/qdma : Integrate qdma pmd to dpdk 22.11

This patch adds Xilinx specific QDMA PCIe functions and
configuration updates to 22.11.
Apply this patch after moving QDMA PMD and test app to respective
directories.

Signed-off-by: Pankaj Darak <pankajd@xilinx.com>
Signed-off-by:  Prasad Pardeshi <prasadp@xilinx.com>
Signed-off-by: Sangani Suryanarayana Raju <ssangani@xilinx.com>
---
 config/meson.build        |  4 ++--
 config/rte_config.h       | 12 +++++++++---
 config/x86/meson.build    |  2 +-
 drivers/net/meson.build   |  1 +
 usertools/dpdk-devbind.py |  6 +++++-
 5 files changed, 18 insertions(+), 7 deletions(-)

diff --git a/config/meson.build b/config/meson.build
index 6d9ffd4..1a7ff90 100644
--- a/config/meson.build
+++ b/config/meson.build
@@ -300,11 +300,11 @@ foreach arg: warning_flags
 endforeach
 
 # set other values pulled from the build options
-dpdk_conf.set('RTE_MAX_ETHPORTS', get_option('max_ethports'))
+dpdk_conf.set('RTE_MAX_ETHPORTS', 256)
 dpdk_conf.set('RTE_LIBEAL_USE_HPET', get_option('use_hpet'))
 dpdk_conf.set('RTE_ENABLE_TRACE_FP', get_option('enable_trace_fp'))
 # values which have defaults which may be overridden
-dpdk_conf.set('RTE_MAX_VFIO_GROUPS', 64)
+dpdk_conf.set('RTE_MAX_VFIO_GROUPS', 256)
 dpdk_conf.set('RTE_DRIVER_MEMPOOL_BUCKET_SIZE_KB', 64)
 dpdk_conf.set('RTE_LIBRTE_DPAA2_USE_PHYS_IOVA', true)
 if dpdk_conf.get('RTE_ARCH_64')
diff --git a/config/rte_config.h b/config/rte_config.h
index 3c4876d..c412749 100644
--- a/config/rte_config.h
+++ b/config/rte_config.h
@@ -34,11 +34,11 @@
 #define RTE_MAX_MEM_MB_PER_LIST 32768
 #define RTE_MAX_MEMSEG_PER_TYPE 32768
 #define RTE_MAX_MEM_MB_PER_TYPE 65536
-#define RTE_MAX_MEMZONE 2560
+#define RTE_MAX_MEMZONE 20480
 #define RTE_MAX_TAILQ 32
 #define RTE_LOG_DP_LEVEL RTE_LOG_INFO
 #define RTE_BACKTRACE 1
-#define RTE_MAX_VFIO_CONTAINERS 64
+#define RTE_MAX_VFIO_CONTAINERS 256
 
 /* bsd module defines */
 #define RTE_CONTIGMEM_MAX_NUM_BUFS 64
@@ -55,7 +55,7 @@
 #define RTE_PKTMBUF_HEADROOM 128
 
 /* ether defines */
-#define RTE_MAX_QUEUES_PER_PORT 1024
+#define RTE_MAX_QUEUES_PER_PORT 2048
 #define RTE_ETHDEV_QUEUE_STAT_CNTRS 16 /* max 256 */
 #define RTE_ETHDEV_RXTX_CALLBACKS 1
 #define RTE_MAX_MULTI_HOST_CTRLS 4
@@ -131,6 +131,12 @@
 /* QEDE PMD defines */
 #define RTE_LIBRTE_QEDE_FW ""
 
+#define RTE_LIBRTE_QDMA_DEBUG_DRIVER 1
+
+#define RTE_LIBRTE_QDMA_PMD 1
+
+#define CONFIG_RTE_LIBRTE_QDMA_PMD 1
+
 /* DLB2 defines */
 // RTE_LIBRTE_PMD_DLB2_QUELL_STATS is not set
 
diff --git a/config/x86/meson.build b/config/x86/meson.build
index 54345c4..3fb74ac 100644
--- a/config/x86/meson.build
+++ b/config/x86/meson.build
@@ -71,5 +71,5 @@ else
 endif
 
 dpdk_conf.set('RTE_CACHE_LINE_SIZE', 64)
-dpdk_conf.set('RTE_MAX_LCORE', 128)
+dpdk_conf.set('RTE_MAX_LCORE', 256)
 dpdk_conf.set('RTE_MAX_NUMA_NODES', 32)
diff --git a/drivers/net/meson.build b/drivers/net/meson.build
index 6470bf3..1d31b2c 100644
--- a/drivers/net/meson.build
+++ b/drivers/net/meson.build
@@ -51,6 +51,7 @@ drivers = [
         'pcap',
         'pfe',
         'qede',
+        'qdma',
         'ring',
         'sfc',
         'softnic',
diff --git a/usertools/dpdk-devbind.py b/usertools/dpdk-devbind.py
index 4d9c1be..9c9d5bc 100755
--- a/usertools/dpdk-devbind.py
+++ b/usertools/dpdk-devbind.py
@@ -30,6 +30,10 @@
               'SVendor': None, 'SDevice': None}
 cavium_pkx = {'Class': '08', 'Vendor': '177d', 'Device': 'a0dd,a049',
               'SVendor': None, 'SDevice': None}
+xilinx_qdma_pf = {'Class':  '05', 'Vendor': '10ee', 'Device': '9011,9111,9211,9311,9014,9114,9214,9314,9018,9118,9218,9318,901f,911f,921f,931f,9021,9121,9221,9321,9024,9124,9224,9324,9028,9128,9228,9328,902f,912f,922f,932f,9031,9131,9231,9331,9034,9134,9234,9334,9038,9138,9238,9338,903f,913f,923f,933f,9041,9141,9241,9341,9044,9144,9244,9344,9048,9148,9248,9348,b011,b111,b211,b311,b014,b114,b214,b314,b018,b118,b218,b318,b01f,b11f,b21f,b31f,b021,b121,b221,b321,b024,b124,b224,b324,b028,b128,b228,b328,b02f,b12f,b22f,b32f,b031,b131,b231,b331,b034,b134,b234,b334,b038,b138,b238,b338,b03f,b13f,b23f,b33f,b041,b141,b241,b341,b044,b144,b244,b344,b048,b148,b248,b348,b058,b158,b258,b358',
+'SVendor': None, 'SDevice': None}
+xilinx_qdma_vf = {'Class':  '05', 'Vendor': '10ee', 'Device': 'a011,a111,a211,a311,a014,a114,a214,a314,a018,a118,a218,a318,a01f,a11f,a21f,a31f,a021,a121,a221,a321,a024,a124,a224,a324,a028,a128,a228,a328,a02f,a12f,a22f,a32f,a031,a131,a231,a331,a034,a134,a234,a334,a038,a138,a238,a338,a03f,a13f,a23f,a33f,a041,a141,a241,a341,a044,a144,a244,a344,a048,a148,a248,a348,c011,c111,c211,c311,c014,c114,c214,c314,c018,c118,c218,c318,c01f,c11f,c21f,c31f,c021,c121,c221,c321,c024,c124,c224,c324,c028,c128,c228,c328,c02f,c12f,c22f,c32f,c031,c131,c231,c331,c034,c134,c234,c334,c038,c138,c238,c338,c03f,c13f,c23f,c33f,c041,c141,c241,c341,c044,c144,c244,c344,c048,c148,c248,c348,c058,c158,c258,c358',
+'SVendor': None, 'SDevice': None}
 cavium_tim = {'Class': '08', 'Vendor': '177d', 'Device': 'a051',
               'SVendor': None, 'SDevice': None}
 cavium_zip = {'Class': '12', 'Vendor': '177d', 'Device': 'a037',
@@ -75,7 +79,7 @@
 virtio_blk = {'Class': '01', 'Vendor': "1af4", 'Device': '1001,1042',
                     'SVendor': None, 'SDevice': None}
 
-network_devices = [network_class, cavium_pkx, avp_vnic, ifpga_class]
+network_devices = [network_class, cavium_pkx, avp_vnic, ifpga_class, xilinx_qdma_pf, xilinx_qdma_vf]
 baseband_devices = [acceleration_class]
 crypto_devices = [encryption_class, intel_processor_class]
 dma_devices = [cnxk_dma, hisilicon_dma,
-- 
2.27.0

